src/pywrangler/sync.py

```diff
@@ -0,0 +1,267 @@
+                "-r",
+                str(GENERATED_REQUIREMENTS_PATH),
+            ]
+        )
+        logger.info(
+            f"Packages installed in [bold]{vendor_path_relative}[/bold].",
+            extra={"markup": True},
+        )
+    else:
+        logger.warning(
+            f"{GENERATED_REQUIREMENTS_PATH} is empty or was not created. No dependencies to install in {vendor_path}."
+        )
+
+    # Create a requirements file for .venv-workers that includes webtypy
+    VENV_REQUIREMENTS_PATH.parent.mkdir(parents=True, exist_ok=True)
+
+    # Start with existing requirements (if any)
+    if (
+        GENERATED_REQUIREMENTS_PATH.is_file()
+        and GENERATED_REQUIREMENTS_PATH.stat().st_size > 0
+    ):
+        with open(GENERATED_REQUIREMENTS_PATH, "r") as src, open(
+            VENV_REQUIREMENTS_PATH, "w"
+        ) as dest:
+            dest.write(src.read())
+    else:
+        # Create a new empty requirements file
+        with open(VENV_REQUIREMENTS_PATH, "w") as f:
+            pass
+
+    # Add webtypy to the venv requirements file for type hints
+    with open(VENV_REQUIREMENTS_PATH, "r") as f:
+        current_content = f.read()
+
+    with open(VENV_REQUIREMENTS_PATH, "a") as f:
+        if current_content and not current_content.endswith("\n"):
+            f.write("\n")
+        f.write("webtypy\n")
+
+    # Install packages into .venv-workers so that user's IDE can see the packages.
+    venv_bin_path = VENV_WORKERS_PATH / ("Scripts" if os.name == "nt" else "bin")
+    venv_python_executable = venv_bin_path / (
+        "python.exe" if os.name == "nt" else "python"
+    )
+
+    # For nicer logs, output the relative path.
+    relative_venv_workers_path = VENV_WORKERS_PATH.relative_to(PROJECT_ROOT)
+    if venv_python_executable.is_file():
+        logger.info(
+            f"Installing packages into [bold]{relative_venv_workers_path}[/bold] using uv pip...",
+            extra={"markup": True},
+        )
+        run_command(
+            [
+                "uv",
+                "pip",
+                "install",
+                "-p",
+                str(venv_python_executable),
+                "-r",
+                str(VENV_REQUIREMENTS_PATH),
+            ]
+        )
+        logger.info(
+            f"Packages installed in [bold]{relative_venv_workers_path}[/bold].",
+            extra={"markup": True},
+        )
+    else:
+        logger.warning(
+            f"Python executable not found at {venv_python_executable}. Skipping installation in [bold]{relative_venv_workers_path}[/bold].",
+            extra={"markup": True},
+        )
+
+    # Clean up temporary files
+    if GENERATED_REQUIREMENTS_PATH.exists():
+        GENERATED_REQUIREMENTS_PATH.unlink()
+        logger.debug(f"Cleaned up {GENERATED_REQUIREMENTS_PATH}.")
+    if VENV_REQUIREMENTS_PATH.exists():
+        VENV_REQUIREMENTS_PATH.unlink()
+        logger.debug(f"Cleaned up {VENV_REQUIREMENTS_PATH}.")
+
+
+def check_timestamps():
```

Comment:
Maybe a better name would be:
```suggestion
def is_sync_needed():
```